File: PPBadgeViewObjc/PPBadgeViewObjc/Classes/Views/PPTableViewCell.m
//
//  PPTableViewCell.m
//  PPBadgeViewObjc
//
//  Created by AndyPang on 2017/6/17.
//  Copyright © 2017年 AndyPang. All rights reserved.
//

#import "PPTableViewCell.h"
#import "PPBadgeView.h"

#define kRandomColor [UIColor colorWithRed:arc4random()%255/255.f green:arc4random()%255/255.f blue:arc4random()%255/255.f alpha:1]


@interface PPTableViewCell ()
@property (weak, nonatomic) IBOutlet UIView *headerView;

@end

@implementation PPTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.headerView.layer.cornerRadius = 3;
    
    [self setupBadge];
}

- (void)setupBadge
{
    // 1. 给headeView添加小圆点
    [self.headerView pp_addDotWithColor:kRandomColor];
    [self.headerView pp_setBadgeFlexMode:PPBadgeViewFlexModeHead];
    [self.headerView pp_moveBadgeWithX:-1 Y:1];
}

@end

File: PPBadgeViewSwift/PPBadgeViewSwift/Classes/Views/PPTableViewCell.swift
//
//  PPTableViewCell.swift
//  PPBadgeViewSwift
//
//  Created by AndyPang on 2017/6/19.
//  Copyright © 2017年 AndyPang. All rights reserved.
//

import UIKit

class PPTableViewCell: UITableViewCell {

    @IBOutlet weak var headerView: UIView!
    
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.headerView.layer.cornerRadius = 3
        self.setupBadge()
    }

    func setupBadge() {
        
        // 1. 给headeView添加小圆点
        self.headerView.pp.addDot(color: randomColor())
        self.headerView.pp.moveBadge(x: -1, y: 1)
    }
    
    
    /// 获取随机颜色
    func randomColor() -> UIColor {
        let randomRed = CGFloat(arc4random()%255)/255.0
        let randomGreen = CGFloat(arc4random()%255)/255.0
        let randomBlue = CGFloat(arc4random()%255)/255.0
        return UIColor.init(red: randomRed, green: randomGreen, blue: randomBlue, alpha: 1.0)
    }
    
}
